gears/src/types/base/coins/decimal.rs

```diff
@@ -79,7 +83,12 @@ impl DecimalCoins {
 
     pub fn checked_sub(&self, other: &DecimalCoins) -> Result<Self, CoinsError> {
         if self.is_all_gte(other.inner()) {
-            let coins = self.checked_calculate_iterate(other.inner(), Decimal256::checked_sub)?;
+            let coins: Vec<DecimalCoin> = self
```

Comment:
Looks like we need documentation and tests around these methods to define the behaviour. Especially since it looks like the behaviour is changed as part of this PR.

Reply:
The logic is covered by tests and added some documentation.